components/suggest/src/store.rs

```diff
@@ -71,31 +100,22 @@ pub struct SuggestIngestionConstraints {
     pub max_suggestions: Option<u64>,
 }
 
-impl SuggestStore {
-    /// Creates a Suggest store.
-    pub fn new(
-        path: &str,
-        settings_config: Option<RemoteSettingsConfig>,
-    ) -> SuggestApiResult<Self> {
-        Ok(Self::new_inner(path, settings_config)?)
-    }
+/// The implementation of the store. This is generic over the Remote Settings
+/// client, and is split out from the concrete [`SuggestStore`] for testing
+/// with a mock client.
+pub(crate) struct GenericSuggestStore<S> {
```

Comment:
Do we want to bound it?
```suggestion
pub(crate) struct GenericSuggestStore<S: SuggestRemoteSettingsClient> {
```

Reply:
Good question about adding a bound to the type declaration! I think the Rust convention leans away from doing that (rust-lang/rust-clippy#1689), and prefers adding the bound to just the `impl` blocks that need it. I don't feel strongly about it, though, and would be happy to add the bound if you'd like—the issues with bounds on the type mentioned in that Clippy ticket aren't as relevant here.

You're right about `Generic` being redundant; I had a hard time thinking of a good name! Let's do `SuggestStoreInner` 😁

Reply:
Yep, that Clippy issue is still open; there's no lint for it or anything now! 👀 But it did help me understand why trait bounds aren't as commonly used, and where they're needed (the `Cow` 🐮 example, where we need the bound to express the `Owned` variant).